config/routes.rb

```diff
@@ -1,31 +1,36 @@
 Rails.application.routes.draw do
-  root 'store#index', as: 'store_index'
+  scope constraints: -> (req) { req.headers['User-Agent'] !~ FIREFOX_BROWSER_REGEX } do
```

Comment:
instead of scope, in case of browser match
you could have used 
match '*path', to: redirect('404'), via: :all

Reply:
I misinterpreted the requirements
```Add a constraint that only home page can be accessed from Firefox. No other page can be seen on firefox. If somebody tries access any other page from Firefox then default rails “no route found” should come```
Will fix it accordingly

Reply:
Fix as metioned in Extension